packages/amd-loader/src/amd-loader.ts

```diff
@@ -0,0 +1,237 @@
+type modCallback = (...args: {}[]) => void;
+
+interface Window {
+  define: (deps: string[], factory: modCallback) => void;
+}
+
+(function() {
```

Comment:
you shouldn't need an IIFE in the source - either the module setting in tsconfig will create one, or just use a block, which should be compiled to an IIFE.

Reply:
The default module setting is CommonJS, which might not be obvious in the output here because you have no imports or exports. `"module": "none"` and using a block might do it.

Reply:
Nope, module: none and target:es5 with a block will just emit a block with `var`s:

tsconfig.json
```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "none"
  }
}
```

test.ts
```ts
{
  let x = 0;
}
```

test.js
```js
{
  var x = 0;
}
```

Reply:
Wow, that just seems broken... unless the rationale is without a module setting they can't know that this file _isn't_ the whole program?

Ok, then. Thanks for humoring me! 🤷‍♂️